src/plugins/query_enhancements/public/datasets/s3_type.ts

```diff
@@ -197,11 +198,10 @@ const fetchDataSources = async (client: SavedObjectsClientContract): Promise<Dat
     perPage: 10000,
   });
   const dataSources: DataStructure[] = resp.savedObjects
-    .filter(
-      (savedObject) =>
-        typeof savedObject.attributes?.dataSourceEngineType === 'string' &&
-        !savedObject.attributes?.dataSourceEngineType?.includes('OpenSearch Serverless')
-    )
+    .filter((savedObject) => {
+      const coercedVersion = semver.coerce(savedObject.attributes.dataSourceVersion);
+      return coercedVersion ? semver.satisfies(coercedVersion, '>=1.0.0') : false;
+    })
```

Comment:
should we use `semver.vaild(savedObject.attributes.dataSourceVersion)` to check datsource version is valid first and then do satisfies check?

Reply:
+1, If semver.coerce returns null, this will lead to semver.satisfies(null, '>=x.x.x'), could cause an issue? You may want to add a null check before using semver.satisfies, as semver.satisfies doesn't handle the null

Reply:
Doesn't the truthy check before satisfies handle your concern?

Reply:
Yes. truthy check should handle the concerns about `null` and `undefined`, should we validate the datasource version is valid? Thanks

Reply:
It's my understanding that if the `dataSourceVersion` is not valid in our situation, then [`semver.coerce()`](https://docs.npmjs.com/cli/v6/using-npm/semver#coercion) should return null.
> Only text which lacks digits will fail coercion

 `dataSourceVersion` is "invalid" when it is an empty string and gets caught by the truthy check.